lib/cocoapods/downloader/request.rb

```diff
@@ -21,7 +21,7 @@ class Request
 
       # @return [Hash<Symbol, String>] The download parameters for this request.
       #
-      attr_reader :params
+      attr_accessor :params
```

Comment:
I was torn on this. I agree that adding mutation here isn't the nicest thing. But it seemed weird to me that the request instance that's held by the user of the downloader would stop being "truthful". Looking at the call sites for `Downloder.download_request` the request is never even saved to a variable so I think I'll change it to be immutable again.